panic when writing to tsi log file

Go a segv with v1.7.9.
It seems the activeLogFile is used without locking.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x525b40]

goroutine 16285375 [running]:
bufio.(*Writer).Available(...)
    /usr/local/go/src/bufio/bufio.go:607
bufio.(*Writer).Write(0x0, 0xc014c46b60, 0xb, 0x10, 0xc014c46b60, 0xb, 0x10)
    /usr/local/go/src/bufio/bufio.go:617 +0x70
github.com/influxdb/tsdb/index/tsi1.(*LogFile).appendEntry(0xc00905e3c0, 0xc06767d7a8, 0xc00905e3c0, 0x8)
    /go/src/github.com/influxdb/tsdb/index/tsi1/log_file.go:601 +0xad
github.com/influxdb/tsdb/index/tsi1.(*LogFile).DeleteSeriesID(0xc00905e3c0, 0x10186c, 0x0, 0x0)
    /go/src/github.com/influxdb/tsdb/index/tsi1/log_file.go:572 +0xcf
github.com/influxdb/tsdb/index/tsi1.(*Partition).DropSeries(0xc0006105a0, 0x10186c, 0x79, 0xecce3bafe073e808)
    /go/src/github.com/influxdb/tsdb/index/tsi1/partition.go:675 +0x3d
github.com/influxdb/tsdb/index/tsi1.(*Index).DropSeries(0xc020fba870, 0x10186c, 0xc06f611a00, 0x75, 0x79, 0x0, 0x0, 0xc01cda7000)
    /go/src/github.com/influxdb/tsdb/index/tsi1/index.go:788 +0x9a
github.com/influxdb/tsdb/engine/tsm1.(*Engine).deleteSeriesRange(0xc0004be240, 0xc170084000, 0x65d46, 0x72955, 0x8000000000000000, 0x7fffffffffffffff, 0x8000000000000000, 0x7fffffffffffffff)
    /go/src/github.com/influxdb/tsdb/engine/tsm1/engine.go:1663 +0xb98
github.com/influxdb/tsdb/engine/tsm1.(*Engine).DeleteSeriesRangeWithPredicate(0xc0004be240, 0x16ca7a0, 0xc00f6056a0, 0xc06767dd40, 0x0, 0x0)
    /go/src/github.com/influxdb/tsdb/engine/tsm1/engine.go:1452 +0x425
github.com/influxdb/tsdb/engine/tsm1.(*Engine).DeleteSeriesRange(0xc0004be240, 0x16ca7a0, 0xc00f6056a0, 0x8000000000000000, 0x7fffffffffffffff, 0x0, 0x0)
    /go/src/github.com/influxdb/tsdb/engine/tsm1/engine.go:1363 +0x7e
github.com/influxdb/tsdb/engine/tsm1.(*Engine).DeleteMeasurement(0xc0004be240, 0xc065e3f200, 0x10, 0x10, 0x0, 0x0)
    /go/src/github.com/influxdb/tsdb/engine/tsm1/engine.go:1771 +0x227
github.com/influxdb/tsdb.(*Shard).DeleteMeasurement(0xc020f5a280, 0xc065e3f200, 0x10, 0x10, 0x10, 0x10)
    /go/src/github.com/influxdb/tsdb/shard.go:744 +0x84
github.com/influxdb/tsdb.(*Store).DeleteMeasurement.func1(0xc020f5a280, 0x0, 0x0)
    /go/src/github.com/influxdb/tsdb/store.go:975 +0x1fe
github.com/influxdb/tsdb.(*Store).walkShards.func1(0xc06102e930, 0xc02883c660, 0xc020f5a280)
    /go/src/github.com/influxdb/tsdb/store.go:1023 +0x48
created by github.com/influxdb/tsdb.(*Store).walkShards
    /go/src/github.com/influxdb/tsdb/store.go:1022 +0x80
~
```

--- a/tsdb/index/tsi1/partition.go
+++ b/tsdb/index/tsi1/partition.go
@@ -626,7 +626,11 @@ func (p *Partition) DropMeasurement(name []byte) error {
 			} else if elem.SeriesID == 0 {
 				break
 			}
-			if err := p.activeLogFile.DeleteSeriesID(elem.SeriesID); err != nil {
+			if err := func() error {
+				p.mu.RLock()
+				defer p.mu.RUnlock()
+				return p.activeLogFile.DeleteSeriesID(elem.SeriesID)
+			}(); err != nil {
 				return err
 			}
 		}
@@ -687,7 +691,11 @@ func (p *Partition) createSeriesListIfNotExists(names [][]byte, tagsSlice []mode
 
 func (p *Partition) DropSeries(seriesID uint64) error {
 	// Delete series from index.
-	if err := p.activeLogFile.DeleteSeriesID(seriesID); err != nil {
+	if err := func() error {
+		p.mu.RLock()
+		defer p.mu.RUnlock()
+		return p.activeLogFile.DeleteSeriesID(seriesID)
+	}(); err != nil {
 		return err
 	}
 

